lib/utils/link.js

```diff
@@ -25,6 +25,24 @@ function linkIfExists (from, to, gently, cb) {
   })
 }
 
+function resolveIfSymlink (maybeSymlinkPath, cb) {
+  fs.lstat(path.resolve(maybeSymlinkPath), function (err, stat) {
+    if (err) return cb.apply(this, arguments)
+    if (!stat.isSymbolicLink()) return cb(null, maybeSymlinkPath)
+    fs.readlink(path.resolve(maybeSymlinkPath), cb)
+  })
+}
+
+function ensureFromIsNotSource (from, to, cb) {
+  resolveIfSymlink(from, function (err, from) {
```

Comment:
Please give the second from a different name, so it's clear what we're talking about here. Maybe `fromDestination`? or something like that?